### PR TITLE
Add support for name, pkgs and diff_attr parameters to zypperpkg.upgrade()/yumpkg.upgrade() - backport 3004

### PR DESCRIPTION
**Pull request by agraul**
### What does this PR do?

Add `diff_attr` to `yumpkg.upgrade()` and add it, together with `name` and `pkgs`, to `zypperpkg.upgrade()`.
Backport of https://github.com/saltstack/salt/pull/62033

### CODE REVIEW

File: tests/pytests/unit/modules/test_zypperpkg.py [hunk 1]
```diff
@@ -254,3 +268,263 @@ def zypper_search_mock(name, *_args, **_kwargs):
...
+    allowvendorchange_param,
+    novendorchange_param,
+    expected,
+):
+    _zypper = zypper._Zypper()
+    _zypper.inst_avc = inst_avc
+    _zypper.dup_avc = dup_avc
+    _zypper.avc = avc
+    _zypper.allow_vendor_change(allowvendorchange_param, novendorchange_param)
+    assert _zypper.avc == expected
```
Comment (SchoolGuy) [inline tests/pytests/unit/modules/test_zypperpkg.py:330]: Do we need to upstream those tests? I don't see them in the upstream PR.

File: tests/pytests/unit/modules/test_zypperpkg.py [hunk 2]
```diff
@@ -254,3 +268,263 @@ def zypper_search_mock(name, *_args, **_kwargs):
...
+
+        zypper.__zypper__.dup_avc = dup_avc
+        zypper.upgrade(
+            dist_upgrade=True,
+            allowvendorchange=allowvendorchange_param,
+            novendorchange=novendorchange_param,
+        )
+        cmd_run_mock.assert_any_call(
+            expected_cmd, output_loglevel="trace", python_shell=False, env={}
+        )
```
Comment (SchoolGuy) [inline tests/pytests/unit/modules/test_zypperpkg.py:467]: Same thing here. Do we need to upstream them?

**Review by SchoolGuy — APPROVED** [review]
LGTM with small questions regarding upstreaming converted tests.

File: tests/pytests/unit/modules/test_zypperpkg.py (hunk 1 above)
Comment (agraul) [reply tests/pytests/unit/modules/test_zypperpkg.py:330]: We need to push those to either https://github.com/saltstack/salt/pull/60421 or https://github.com/saltstack/salt/pull/62033, depends on which will be merged first.

File: tests/pytests/unit/modules/test_zypperpkg.py (hunk 2 above)
Comment (agraul) [reply tests/pytests/unit/modules/test_zypperpkg.py:467]: See above: yes either in https://github.com/saltstack/salt/pull/62033 or in https://github.com/saltstack/salt/pull/60421

**Pull request merged**